internal/grpc/services/gateway/storageprovider.go

```diff
@@ -440,6 +1015,274 @@ func (s *svc) ListContainer(ctx context.Context, req *provider.ListContainerRequ
+			panic("gateway: a share name must be of type reference: ref:" + res.Info.Path)
+		}
+
+		ri, err := s.checkRef(ctx, res.Info)
+		if err != nil {
+			return &provider.ListContainerResponse{
+				Status: status.NewInternal(ctx, err, "gateway: error resolving reference:"+p),
+			}, nil
+		}
+
+		if ri.Type != provider.ResourceType_RESOURCE_TYPE_CONTAINER {
+			err := errtypes.NotSupported("gateway: list container: cannot list non-container type:" + ri.Path)
+			log.Err(err).Msg("gateway: error listing")
+			return &provider.ListContainerResponse{
+				Status: status.NewInvalidArg(ctx, "resource is not a container"),
+			}, nil
+		}
+
+		target := path.Join(ri.Path, shareChild)
+		ref = &provider.Reference{
+			Spec: &provider.Reference_Path{
+				Path: target,
+			},
+		}
+
+		newReq := &provider.ListContainerRequest{Ref: ref}
+		newRes, err := s.listContainer(ctx, newReq)
+		if err != nil {
+			return &provider.ListContainerResponse{
+				Status: status.NewInternal(ctx, err, "gateway: error listing"),
+			}, nil
+		}
+
+		if newRes.Status.Code != rpc.Code_CODE_OK {
+			err := status.NewErrorFromCode(newRes.Status.Code, "gateway")
+			log.Err(err).Msg("gateway: error listing")
+			return &provider.ListContainerResponse{
+				Status: status.NewInternal(ctx, err, "gateway: error listing"),
+			}, nil
+		}
+
+		// paths needs to be converted
+		for _, info := range newRes.Infos {
+			base := path.Base(info.Path)
+			info.Path = path.Join(shareName, shareChild, base)
+		}
+
+		return newRes, nil
+
+	}
+
+	panic("gateway: stating an unknown path:" + p)
+}
+
+func (s *svc) getPath(ctx context.Context, ref *provider.Reference) (string, error) {
+	if ref.GetPath() != "" {
+		return ref.GetPath(), nil
+	}
+
+	if ref.GetId() != nil {
+		req := &provider.StatRequest{Ref: ref}
+		res, err := s.stat(ctx, req)
+		if err != nil {
+			err = errors.Wrap(err, "gateway: error stating ref:"+ref.String())
+			return "", err
+		}
+
+		if res.Status.Code != rpc.Code_CODE_OK {
+			err := status.NewErrorFromCode(res.Status.Code, "gateway")
+			return "", err
+		}
+
+		return res.Info.Path, nil
+	}
+
+	return "", errors.New("gateway: ref is invalid:" + ref.String())
+}
+
+// /home/MyShares/
+func (s *svc) isSharedFolder(ctx context.Context, p string) bool {
+	return s.split(ctx, p, 2)
+}
+
+// /home/MyShares/photos/
+func (s *svc) isShareName(ctx context.Context, p string) bool {
+	return s.split(ctx, p, 3)
+}
+
+// /home/MyShares/photos/Ibiza/beach.png
+func (s *svc) isShareChild(ctx context.Context, p string) bool {
+	return s.split(ctx, p, 4)
+}
+
+// always validate that the path contains the share folder
+// split cannot be called with i<2
+func (s *svc) split(ctx context.Context, p string, i int) bool {
+	log := appctx.GetLogger(ctx)
+	if i < 2 {
+		panic("split called with i < 2")
+	}
+
+	parts := s.splitPath(ctx, p)
+
+	// validate that we have always at least two elements
+	if len(parts) < 2 {
```

Comment:
there might be only one part when stating `/home` ... which happens on every propfind.
```
10:40PM ERR home/jfd/Repositories/reva/internal/grpc/interceptors/recovery/recovery.go:50 > split: len(parts) < 2: path:/home parts:[home]
```

Reply:
can you try with master ? shouldn't happen, listing /home is okay?

Reply:
this is master:
```go
	// validate that we have always at least two elements
	if len(parts) < 2 {
		panic(fmt.Sprintf("split: len(parts) < 2: path:%s parts:%+v", p, parts))
	}
```
The logic does not work out when stating `/home` because trimmed for `/` it is just `home` and that has only one path segment, not two.

Reply:
might be caused by not having set a `share_folder` ... checking

Reply:
ok, i can do propfinds when specifying an explicit `share_folder`